Zend/zend_language_parser.y

```diff
@@ -1007,31 +1008,47 @@ method_body:
 	|	'{' inner_statement_list '}'	{ $$ = $2; }
 ;
 
-variable_modifiers:
-		non_empty_member_modifiers		{ $$ = $1; }
-	|	T_VAR							{ $$ = ZEND_ACC_PUBLIC; }
+property_modifiers:
+		non_empty_member_modifiers
+			{ $$ = zend_modifier_list_to_flags(ZEND_MODIFIER_TARGET_PROPERTY, $1);
+			  if (!$$) { YYERROR; } }
+	|	T_VAR
+			{ $$ = ZEND_ACC_PUBLIC; }
 ;
 
 method_modifiers:
-		%empty						{ $$ = ZEND_ACC_PUBLIC; }
+		%empty
+			{ $$ = ZEND_ACC_PUBLIC; }
+	|	non_empty_member_modifiers
+			{ $$ = zend_modifier_list_to_flags(ZEND_MODIFIER_TARGET_METHOD, $1);
+			  if (!$$) { YYERROR; }
+			  if (!($$ & ZEND_ACC_PPP_MASK)) { $$ |= ZEND_ACC_PUBLIC; } }
+;
+
+class_const_modifiers:
+		%empty
+			{ $$ = ZEND_ACC_PUBLIC; }
 	|	non_empty_member_modifiers
-			{ $$ = $1; if (!($$ & ZEND_ACC_PPP_MASK)) { $$ |= ZEND_ACC_PUBLIC; } }
+			{ $$ = zend_modifier_list_to_flags(ZEND_MODIFIER_TARGET_CONSTANT, $1);
+			  if (!$$) { YYERROR; }
+			  if (!($$ & ZEND_ACC_PPP_MASK)) { $$ |= ZEND_ACC_PUBLIC; } }
 ;
 
 non_empty_member_modifiers:
-		member_modifier			{ $$ = $1; }
+		member_modifier
+			{ $$ = zend_ast_create_list(1, ZEND_AST_MODIFIER_LIST, zend_ast_create_zval_from_long($1)); }
```

Comment:
We can avoid this list by creating separate bitflags for the modifiers instead of using the token id, and then translating those to the ACC flags instead. This list is essentially trash in the arena, although modifier lists are short and so this likely doesn't matter much.

Reply:
Tried this but this results in much duplication which is likely just not worth it.